lib/parser/source/buffer.rb

```diff
@@ -253,14 +253,11 @@ def last_line
 
       def line_begins
         unless @line_begins
-          @line_begins, index = [ [ 0, 0 ] ], 1
-
-          @source.each_char do |char|
-            if char == NEW_LINE
-              @line_begins.unshift [ @line_begins.length, index ]
-            end
+          @line_begins, index = [ [ 0, 0 ] ], 0
 
+          while index = @source.index("\n".freeze, index)
```

Comment:
We have a lot of those sprinkled all over the codebase--we need to fix them either to this new form or the old form. @bbatsov, how much do we care about the versions of Ruby without this optimization?